standardization/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -171,13 +172,17 @@ object StandardizationJob {
     }
   }
 
-  private def getCsvOptions(cmd: CmdConfig): HashMap[String,Option[RawFormatParameter]] = {
+  private def getCsvOptions(cmd: CmdConfig, schema: Option[StructType] = None): HashMap[String,Option[RawFormatParameter]] = {
     if (cmd.rawFormat.equalsIgnoreCase("csv")) {
+      val knownNumberOfFields = if (schema.isDefined) schema.get.fields.length else 0
       HashMap(
         "delimiter" -> cmd.csvDelimiter.map(StringParameter),
         "header" -> cmd.csvHeader.map(BooleanParameter),
         "quote" -> cmd.csvQuote.map(StringParameter),
-        "escape" -> cmd.csvEscape.map(StringParameter)
+        "escape" -> cmd.csvEscape.map(StringParameter),
+        // increase the default limit on the number of columns if needed
+        // default is set at org.apache.spark.sql.execution.datasources.csv.CSVOptions maxColumns
+        "maxColumns" -> Option(LongParameter(max(20480, knownNumberOfFields)))
```

Comment:
Can we extract the number in case it changes over time or we realize something?

Reply:
thnx, extracted as a constant. (see also discussion in the comment below)